tests/PaginateTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Spatie\CollectionMacros\Test;
+
+use Illuminate\Support\Collection;
+
+class PaginateTest extends TestCase
+{
+    public function setUp()
+    {
+        $this->p = (new Collection(['item1', 'item2', 'item3', 'item4']))->paginate(2, 'page', 2, null);
```

Comment:
Thanks, we'll take a more detailed look at the code in the coming days. But on first glance it looks pretty good.